blockchain.go

```diff
@@ -0,0 +1,202 @@
+// Package broadcaster contains code for interacting with the decentr blockchain.
+package broadcaster
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	clicontext "github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+)
+
+// ErrTxInMempoolCache is returned when tx is already broadcast and exists in mempool cache.
+var ErrTxInMempoolCache = errors.New("tx is already in mempool cache")
+
+var errInvalidSequence = errors.New("invalid sequence")
+
+// Broadcaster provides functionality to broadcast messages to cosmos based blockchain node.
+type Broadcaster struct {
+	ctx clicontext.CLIContext
+	enc sdk.TxEncoder
+
+	genesisKeyPass string
+	chainID        string
```

Comment:
```golang
// BuildAndSign builds a single message to be signed, and signs a transaction
// with the built message given a name, passphrase, and a set of messages.
func (bldr TxBuilder) BuildAndSign(name, passphrase string, msgs []sdk.Msg) ([]byte, error) {
	msg, err := bldr.BuildSignMsg(msgs)
	if err != nil {
		return nil, err
	}

	return bldr.Sign(name, passphrase, msg)
}
```
genesisKeyPass => keybasePassphrase

Reply:
It's not a pass phrase for key base, it's a pass phrase for genesis. We use default '12345678'